src/scenes/gameuiscreen.ts

```diff
@@ -1,28 +1,51 @@
 import { ScreenHeight, ScreenWidth } from "core/main";
 import Phaser from "phaser";
 import { log } from "utilities/logger";
-import { interactify } from "utilities/utils";
+import { assert, interactify } from "utilities/utils";
+import { float, int, Nullable } from "../definitions/utils";
+import { Tile } from "../entities/Tile";
+import TileManager from "../managers/TileManager";
 
 export class GameUiScreen extends Phaser.Scene {
+    private back: Nullable<Phaser.GameObjects.Rectangle>;
+
     public constructor() {
         super("GameUiScreen");
+        this.back = null;
+    }
+
+    public preload(): void {
+        for (let i: int = 0; i < 9; i++) {
+            this.load.image(`tile${i}`, `assets/tiles/tile${i}.png`);
+        }
     }
 
     public create(): void {
-        const back: Phaser.GameObjects.Rectangle = this.add.rectangle(
+        this.back = this.add.rectangle(
             ScreenWidth / 2,
             ScreenHeight - 100,
             ScreenWidth * 0.8,
             200,
-            0xffffff,
+            0x808080,
         );
-        back.setInteractive();
+        this.back.setInteractive();
+        this.displayDrawnTiles();
+    }
 
-        const button: Phaser.GameObjects.Image = this.add.image(
-            ScreenWidth / 2,
-            ScreenHeight - 100,
-            "start",
-        );
-        interactify(button, 0.5, () => log("button"));
+    private displayDrawnTiles(): void {
+        const myTiles: Nullable<Tile[]> = TileManager.getTilesInHand();
+        assert(myTiles);
+        const nrTiles: int = myTiles.length;
+        for (let i: int = 0; i < nrTiles; i++) {
+            const tileSpacing: int =
+                (this.back.width - nrTiles * 128) / (nrTiles + 1);
+            this.add.image(
```

Comment:
I believe you should remove the image from screen if you dont wish it to be visible anymore. Like this, I assume that when getting close to end game, there is nothing to draw and after i play something, the new tiles will be added over the old tiles, the old tiles will probably still be visible.

In LobbyScreen, this problem is solved with a Phaser.GameObject.Container.